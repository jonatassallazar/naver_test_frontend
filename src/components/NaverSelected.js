import React, { useContext } from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { TrashIcon, EditIcon, CloseIcon } from './Icon';
import moment from 'moment';
import store from '../store/store';

const NaverModal = styled.div`
  background: ${({ theme }) => theme.colors.bgHover};
  width: 100%;
  height: 100%;
  position: fixed;
  left: 0;
  top: 0;
`;

NaverModal.Content = styled.div`
  width: 80%;
  display: flex;
  flex-direction: row;
  position: fixed;
  z-index: 6;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);

  @media (max-width: ${({ theme }) => theme.mobile}) {
    flex-direction: column;
    width: 100%;
    padding: ${({ theme }) => theme.spacing.medium};
  }
`;

NaverModal.Data = styled.div`
  background: ${({ theme }) => theme.colors.bg};
  width: 50%;
  padding: ${({ theme }) => theme.spacing.medium};

  h4 {
    font-weight: 600;
    font-size: 1.6rem;
    line-height: 2.4rem;
    margin: ${({ theme }) => theme.spacing.small} 0 0 0;
  }

  h5 {
    font-weight: 400;
    font-size: 1.6rem;
    line-height: 2.4rem;
    margin: ${({ theme }) => theme.spacing.small} 0 0 0;
  }

  @media (max-width: ${({ theme }) => theme.mobile}) {
    width: 100%;
    padding: ${({ theme }) => theme.spacing.small} ${({ theme }) => theme.spacing.small} ${({ theme }) => theme.spacing.huge};
  }
`;

NaverModal.Img = styled.img`
  width: 50%;

  @media (max-width: ${({ theme }) => theme.mobile}) {
    width: 100%;
  }
`;

NaverModal.Close = styled.button`
  position: absolute;
  right: 2rem;
  top: 2rem;
  border: none;
  background: none;
  cursor: pointer;

  svg {
    width: 1.4rem;
    height: 1.4rem;
  }

  @media (max-width: ${({ theme }) => theme.mobile}) {
    padding: 0.8rem 0.8rem 0.6rem;
    margin: ${({ theme }) => theme.spacing.small};
    background: white;
    border-radius: 50%;
  }
`;

NaverModal.Name = styled.h2`
  font-weight: 600;
  font-size: 2.4rem;
  line-height: 3.6rem;
  margin: ${({ theme }) => theme.spacing.small} 0 0 0;

  @media (max-width: ${({ theme }) => theme.mobile}) {
    margin: 0;
  }
`;

NaverModal.Role = styled.p`
  font-weight: 400;
  font-size: 1.6rem;
  line-height: 2.4rem;
  margin: ${({ theme }) => theme.spacing.small} 0
    ${({ theme }) => theme.spacing.large} 0;

  @media (max-width: ${({ theme }) => theme.mobile}) {
    margin: 0;
  }
`;

NaverModal.Icons = styled.div`
  position: absolute;
  bottom: 2rem;

  button {
    border: none;
    margin: 0;
    padding: 0;
    background: none;
  }

  svg {
    margin-right: ${({ theme }) => theme.spacing.small};
    color: ${({ theme }) => theme.colors.primary};
    cursor: pointer;
  }

  @media (max-width: ${({ theme }) => theme.mobile}) {
    padding-bottom: ${({ theme }) => theme.spacing.small};
  }
`;

const NaverSelected = ({ naverSelected, setShowNaverModalSelected }) => {
  const { dispatch } = useContext(store);

  const handleDelete = () => {
    dispatch({
      type: 'CONFIRMATION_DELETE_NAVER',
      id: naverSelected.id,
      dispatch,
    });
  };

  const handleClose = () => {
    setShowNaverModalSelected(false);
  };

  return (
    <>
      <NaverModal.Content>
        <NaverModal.Img alt={naverSelected.name} src={naverSelected.url} />
        <NaverModal.Data>
          <NaverModal.Close onClick={handleClose}>
            <CloseIcon />
          </NaverModal.Close>
          <NaverModal.Name>{naverSelected.name}</NaverModal.Name>
          <NaverModal.Role>{naverSelected.job_role}</NaverModal.Role>
          <h4>Idade</h4>
          <h5>{moment(naverSelected.birthdate).format('DD/MM/YYYY')}</h5>
          <h4>Tempo de Empresa</h4>
          <h5>{moment(naverSelected.admission_date).format('DD/MM/YYYY')}</h5>
          <h4>Projetos que participou</h4>
          <h5>{naverSelected.project}</h5>
          <NaverModal.Icons>
            <button onClick={handleDelete}>
              <TrashIcon />
            </button>
            <Link to={`/navers/edit/${naverSelected.id}`}>
              <EditIcon />
            </Link>
          </NaverModal.Icons>
        </NaverModal.Data>
      </NaverModal.Content>
      <NaverModal onClick={handleClose} />
    </>
  );
};

export default NaverSelected;
