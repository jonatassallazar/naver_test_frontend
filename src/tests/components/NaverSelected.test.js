import NaverSelected from '../../components/NaverSelected';
import customRender from './customRender';
import navers from '../fixtures/fixtures';
import 'jest-styled-components';

test('should match LoginPage snapshot', () => {
  const setShowNaverModalSelected = jest.fn();

  const { container } = customRender(
    <NaverSelected
      naverSelected={navers[2]}
      setShowNaverModalSelected={setShowNaverModalSelected}
    />,
  );

  expect(container).toMatchSnapshot();
});
