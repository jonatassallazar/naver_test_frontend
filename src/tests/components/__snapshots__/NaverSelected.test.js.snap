// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match LoginPage snapshot 1`] = `
.c4 {
  width: 1.8rem;
  height: 1.8rem;
}

.c8 {
  background: rgba(0,0,0,0.5);
  width: 100%;
  height: 100%;
  position: fixed;
  left: 0;
  top: 0;
}

.c0 {
  width: 80%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: fixed;
  z-index: 6;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c2 {
  background: #ffffff;
  width: 50%;
  padding: 2.4rem;
}

.c2 h4 {
  font-weight: 600;
  font-size: 1.6rem;
  line-height: 2.4rem;
  margin: 1.4rem 0 0 0;
}

.c2 h5 {
  font-weight: 400;
  font-size: 1.6rem;
  line-height: 2.4rem;
  margin: 1.4rem 0 0 0;
}

.c1 {
  width: 50%;
}

.c3 {
  position: absolute;
  right: 2rem;
  top: 2rem;
  border: none;
  background: none;
  cursor: pointer;
}

.c3 svg {
  width: 1.4rem;
  height: 1.4rem;
}

.c5 {
  font-weight: 600;
  font-size: 2.4rem;
  line-height: 3.6rem;
  margin: 1.4rem 0 0 0;
}

.c6 {
  font-weight: 400;
  font-size: 1.6rem;
  line-height: 2.4rem;
  margin: 1.4rem 0 3.2rem 0;
}

.c7 {
  position: absolute;
  bottom: 2rem;
}

.c7 button {
  border: none;
  margin: 0;
  padding: 0;
  background: none;
}

.c7 svg {
  margin-right: 1.4rem;
  color: #212121;
  cursor: pointer;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    width: 100%;
    padding: 2.4rem;
  }
}

@media (max-width:768px) {
  .c2 {
    width: 100%;
    padding: 1.4rem 1.4rem 6rem;
  }
}

@media (max-width:768px) {
  .c1 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c3 {
    padding: 0.8rem 0.8rem 0.6rem;
    margin: 1.4rem;
    background: white;
    border-radius: 50%;
  }
}

@media (max-width:768px) {
  .c5 {
    margin: 0;
  }
}

@media (max-width:768px) {
  .c6 {
    margin: 0;
  }
}

@media (max-width:768px) {
  .c7 {
    padding-bottom: 1.4rem;
  }
}

<div>
  <div
    class="c0"
  >
    <img
      alt="Tester 3"
      class="c1"
      src="https://lorem.picsum.com"
    />
    <div
      class="c2"
    >
      <button
        class="c3"
      >
        <svg
          class=" c4"
          version="1.1"
          viewBox="0 0 14 14"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M14 1.41L12.59 0L7 5.59L1.41 0L0 1.41L5.59 7L0 12.59L1.41 14L7 8.41L12.59 14L14 12.59L8.41 7L14 1.41Z"
            fill="currentColor"
          />
        </svg>
      </button>
      <h2
        class="c5"
      >
        Tester 3
      </h2>
      <p
        class="c6"
      >
        Backdev
      </p>
      <h4>
        Idade
      </h4>
      <h5>
        05/11/2001
      </h5>
      <h4>
        Tempo de Empresa
      </h4>
      <h5>
        12/01/2014
      </h5>
      <h4>
        Projetos que participou
      </h4>
      <h5>
        all
      </h5>
      <div
        class="c7"
      >
        <button>
          <svg
            class=" c4"
            version="1.1"
            viewBox="0 0 14 18"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <path
              d="M1 18H13V4H1V18ZM14 1H10.5L9.5 0H4.5L3.5 1H0V3H14V1Z"
              fill="currentColor"
            />
          </svg>
        </button>
        <a
          href="/navers/edit/3"
        >
          <svg
            class=" c4"
            version="1.1"
            viewBox="0 0 19 19"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <path
              d="M0 15.25V19H3.75L14.81 7.94L11.06 4.19L0 15.25ZM18.41 4.34L14.66 0.589996L12.13 3.13L15.88 6.88L18.41 4.34Z"
              fill="currentColor"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
  <div
    class="c8"
  />
</div>
`;
